tools/docvalidator/main.go

```diff
@@ -0,0 +1,159 @@
+// Package main provides a utility to process markdown documentation files,
+// extracts code examples, executes them, and verifies their outputs.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+	"slices"
+	"strings"
+	"time"
+)
+
+// Example represents an individual code example extracted from the markdown files.
+// It contains the function name, the file it was extracted from, the code snippet, and the expected output.
+type Example struct {
+	FuncName string
+	File     string
+	Code     string
+	Expected string
+	Skipped  bool
+}
+
+var logLevel string
+
+// init initializes the logging level from command-line flags.
+// It sets the default logging level to "info", but can be overridden by
+// specifying a different level (debug, info, warn, error) using the -log flag.
+func init() {
+	flag.StringVar(&logLevel, "log", "info", "Set the logging level (debug, info, warn, error)")
+}
+
+// main is the entry point of the application that processes markdown documentation files
+// to extract, execute, and verify code examples. It performs the following steps:
+// 1. Parses command-line flags to set the logging level.
+// 2. Sets the log level based on the flag value and forces the timezone to UTC.
+// 3. Discovers markdown files in the 'docs/registries' directory, filtering out specific files.
+// 4. Iterates over each file, extracting code examples and processing them.
+// 5. For each example, logs the processing status and counts errors.
+// 6. Exits with an error code if any examples fail processing, or with success if all pass.
+func main() {
+	// Parse command-line flags
+	flag.Parse()
+
+	// Create an instance of the real file system
+	fs := &OSFileSystem{}
+
+	// Call the run function
+	if err := run(fs); err != nil {
+		// If an error occurs, log it and exit with code 1
+		slog.Error("An error occurred during validation", "error", err)
+		os.Exit(1)
+	}
+	// Exit successfully
+	os.Exit(0)
+}
+
+// run contains the main logic of the program.
+// It processes markdown files, extracts code examples, executes them, and verifies their outputs.
+// This function returns an error if any step fails, making it easier to test.
+func run(fs FileSystem) error {
+	// Set the log level based on the flag
+	if err := setLogLevel(logLevel); err != nil {
+		return err
+	}
+
+	// Force timezone to UTC (for date parsing)
+	time.Local = time.UTC
+
+	// Discover markdown files in the docs directory
+	files, err := fs.Glob("docs/registries/*.md")
```

Comment:
I know some people out there are using windows to code. I'm so sad for them, but you could help them with OS agnostic tests

```suggestion
	files, err := fs.Glob(filepath.Join ("docs", "registries", "*.md")
```